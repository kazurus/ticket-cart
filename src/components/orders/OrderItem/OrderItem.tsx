import React from 'react';
import Order from '../Order';

type OrderProp = {
  from: string;
  to: string;
  price: number;
  id: number;
  createdAt: number;
};

interface Props {
  order: OrderProp;
  onDelete: () => void;
}

const OrderItem: React.FC<Props> = ({ order, onDelete }) => {
  const { id, createdAt, from, to, price } = order;

  return (
    <div style={{ marginBottom: 20 }} data-testid='order-item'>
      <Order {...{ id, createdAt, from, to, price }} />
      <button type='button' data-testid='delete-button' onClick={onDelete}>
        Delete
      </button>
    </div>
  );
};

export default OrderItem;
