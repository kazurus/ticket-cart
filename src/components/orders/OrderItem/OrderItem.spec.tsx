import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import OrderItem from './OrderItem';

const order = {
  id: 1,
  createdAt: 111,
  to: 'to',
  from: 'from',
  price: 10,
};

const onDelete = jest.fn();

describe('OrderItem component', () => {
  test('should render props', () => {
    const { container } = render(
      <OrderItem order={order} onDelete={onDelete} />
    );

    expect(container).toMatchSnapshot();
  });

  test('should call the onDelete callback handler', () => {
    const { getByTestId } = render(
      <OrderItem order={order} onDelete={onDelete} />
    );

    fireEvent.click(getByTestId('delete-button'));

    expect(onDelete).toBeCalledTimes(1);
  });
});
