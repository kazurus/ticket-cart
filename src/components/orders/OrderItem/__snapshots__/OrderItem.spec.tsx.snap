// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem component should render props 1`] = `
<div>
  <div
    data-testid="order-item"
    style="margin-bottom: 20px;"
  >
    <div>
      Order: 
      1
       - 
      111
      <br />
      from
       - 
      to
       $
      10
    </div>
    <button
      data-testid="delete-button"
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
